File: src/main/java/com/emirhan/ft_case/data/repository/CommentRepository.java
package com.emirhan.ft_case.data.repository;

import com.emirhan.ft_case.data.entity.CommentEntity;
import com.emirhan.ft_case.data.entity.ProductEntity;
import com.emirhan.ft_case.data.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface CommentRepository extends JpaRepository<CommentEntity, Integer> {
    Optional<List<CommentEntity>> findAllByProduct(ProductEntity product);

    Optional<List<CommentEntity>> findAllByUser(UserEntity user);
}

File: src/main/java/com/emirhan/ft_case/domain/repository/CommentRepository.kt
package com.emirhan.ft_case.domain.repository

import com.emirhan.ft_case.domain.entity.CommentEntity
import java.util.Date

interface CommentRepository {
    fun addComment(comment: CommentEntity)
    fun getUserComments(id: Int): List<CommentEntity>
    fun getProductComments(id: Int): List<CommentEntity>
    fun getProductCommentsBetweenDates(id: Int, startDate: Date, endDate: Date): List<CommentEntity>
    fun getUserCommentsBetweenDates(id: Int, startDate: Date, endDate: Date): List<CommentEntity>
}